lib/runner.rb

```diff
@@ -0,0 +1,163 @@
+# frozen_string_literal: true
+
+require 'scraperwiki'
+require 'rest-client'
+require 'optparse'
+
+# Wrapper for all runners
+module Jacaranda
+  def self.run(args)
+    parse(args)
+    announce
+    runners.each(&:run)
+  end
+
+  # rubocop:disable Metrics/MethodLength
+  def self.parse(args)
+    opt_parser = OptionParser.new do |opts|
+      opts.banner = "Usage: #{$PROGRAM_NAME} [options]"
+      opts.on('-r', '--runners=RUNNER[,<RUNNER>,...]', 'Runners to execute') do |r|
+        @whitelist = r.split(',')
+      end
+      opts.on('-h', '--help', 'Prints this help') do
+        puts opts
+        exit
+      end
+    end
+    opt_parser.parse!(args)
+  end
+  # rubocop:enable Metrics/MethodLength
+
+  def self.runners
+    @whitelist ||= []
+    candidates = self::Runner.descendants
+    candidates.select! { |c| @whitelist.find { |w| c.to_s =~ /#{w}/i } } unless @whitelist.empty?
+    candidates.sort_by { |c| c.to_s.split('::').last }
+  end
+
+  def self.announce
+    puts 'These are the runners we will execute:'
+    puts
+    puts runners.map { |r| r.to_s.split('::').last }.join("\n")
+    puts
+    sleep(2)
+  end
+
+  # Boilerplate for running stat scrapers
+  class Runner
+    class << self
+      def run
+        validate_environment_variables!
+
+        if posted_in_last_fortnight?
+          puts "[#{name}] We have posted an update during this fortnight."
+          false
+        else
+          puts "[#{name}] We have not posted an update during this fortnight."
+          scrape_and_post_message
+          true
+        end
+      end
+
+      def required_environment_variables
+        %w[MORPH_LIVE_MODE MORPH_SLACK_CHANNEL_WEBHOOK_URL]
+      end
+
+      def validate_environment_variables!
+        return if required_environment_variables.all? { |var| ENV[var] }
+
+        puts "[#{name}] The runner needs the following environment variables set:"
+        puts
+        puts required_environment_variables.join("\n")
+        exit(1)
+      end
+
+      def posts
+        query = "* from data where runner = '#{self}'"
+        posts = ScraperWiki.select(query)
+        posts.map do |post|
+          post['date_posted'] = Date.parse(post['date_posted'])
```

Comment:
This could be extracted to a named method I recon, it wasn't obvious at first this is just parsing the dates into Ruby Date objects and returning them.